client/src/components/layouts/default.tsx

```diff
@@ -94,13 +94,6 @@ interface DefaultLayoutProps extends StateProps, DispatchProps {
   t: TFunction;
 }
 
-const getSystemTheme = () =>
-  `${
-    window.matchMedia('(prefers-color-scheme: dark)').matches === true
-      ? 'dark-palette'
-      : 'light-palette'
-  }`;
```

Comment:
Why was this removed? I thought the point of this was to allow us to detect the user preferences without having to duplicate the css variables a bunch of times.

Reply:
I should have provided content about my meeting with Ahmad, why we decided to keep it in our meeting before holiday. Sorry about it.

## How to solve the issue with JavaScript

If we keep the behavior, then we have to add:

- check if there is `currentTheme`, to apply the system theme
- check after the user logged in, to apply the JavaScript code

Both are doable but not common, but it feels instead of applying the theme to the canvas, you are changing the canvas to fit a theme, which isn't mainstreamed anymore, instead in the mainstream people let the browser provide the user with desired output without hassle.

### Issues with using JavaScript

Although, this's okay for our codebase, because we are using outdated libraries, that don't use `preferred-color-scheme` in them, like in editor, more details in issue #48936.

But if we are going to move forward and adapt more modern and easier solution that allow for us more options, like the UI Component library, that have components (one of them is FormGroup #46747, it always comes to that component 😨) that use `preferred-color-scheme` or other libraries.

Second point is: it's harder to find the issue, when people are used to a different system. Meaning that people who are used to find the theme colors and its styles in one file called `theme.css`, won't expect us to be using JavaScript to set the theme, so it's harder for people who are new and following the standard of the industry to manipulate or fix the themes issues.

---

## But what about issue with variables based on my discussion with Ahmad?

- It will clutter the devtools, which make it harder for developers to manipulate values in the browser.

With the usage of the selectors (if I did it right, I haven't checked because I was focused on the buttons), the variables will be hidden.

Reply:
As I understand it, the plan was to keep the setting in local storage and update it whenever the user data is fetched. It's not a particularly convoluted flow.

We can put the all themes in `theme.css`, that's fine.  I just don't think we should be duplicating them.

Also, setting the theme via an attribute on the html element is pretty common. Reddit, GitHub and the React docs all do it and they're about as mainstream as it gets.

Reply:
Setting in local storage is related to the navigate bar to change the theme without logging in, i.e. someone wants to change the theme without changing the whole preferred color scheme in their browser.

Using local storage in the setting is counter-intuitive because they are logged in and connected to the database.

---

If you mean that instead of using the values to database, we use the values in the local-storage when changing to system, this will add a can of worms of how to revert to the `preferred-color-scheme`, when the local storage value is filled when they are offline.

---

Yes, using attribute after deciding the desired theme is common, I am not deleting it. I am respecting the browser behavior of allowing it to follow the user preference, which is in GitHub, React docs and Mozilla docs

Picture from GitHub variables, in incognito mode

![image](https://user-images.githubusercontent.com/88248797/211043920-b9aac4b7-285a-425e-bdc5-74922982d30d.png)

---

> I just don't think we should be duplicating them. 

True, maybe we can remove a selector or two. but I haven't tested (pushed the limits with the selector yet, was focused on apply the opt in system). I may not be able to do it, t.

Reply:
Fair point. I think GitHub gets away with it by using SASS to define each theme once and then use them repeatedly in the various media queries.

All I'm trying to say is that defining a theme more than once is not a good idea. Even if it doesn't cause weird bugs, it will make maintenance harder.

Reply:
#we-should-use-sassOrtailwind, I do see the point, don't worry I haven't actively tried to refactor it, so there maybe a way to do it in this PR, that doesn't require remaking our theming system that I have in mind

Reply:
this is the minimal selectors that I can think of, it doesn't use `:not` selector in `:root` so the :root is clutter in the devtools instead of the previous way.

Reply:
That's a neat solution, nice work.  I was worried we'd have to use SASS for a second, there :cold_sweat:

Reply:
Just a heads up that whatever we end up deciding here should be reflected in the Component library's theme system.